extensions/oidc/runtime/src/main/java/io/quarkus/oidc/SessionManager.java

```diff
@@ -0,0 +1,15 @@
+package io.quarkus.oidc;
+
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Authorization Code Flow Session Manager
+ */
+public interface SessionManager {
+
+    String createSessionState(RoutingContext routingContext, OidcTenantConfig oidcConfig, SessionContent sessionContent);
```

Comment:
Perhaps `SessionManager` should be solely responsible for managing cookies and this method just returns `void`. My point is that we remove any session cookie related code from the mechanism and move here.

I would then just rename these methods to `create`, `get` and `delete`.

Reply:
@pedroigor As I've tried to justify, it is actually a `SessionManager` :-)